internal/transport/http2_client.go

```diff
@@ -863,6 +863,13 @@ func (t *http2Client) Close() error {
 		// should unblock it so that the goroutine eventually exits.
 		t.kpDormancyCond.Signal()
 	}
+	// The HTTP/2 spec mentions that a GOAWAY frame should be sent before a connection close. If this close() function
+	// ever starts to take in an HTTP/2 error code the peer will be able to get more information about the reason
+	// behind the connection close.
+	if err := t.framer.fr.WriteGoAway(math.MaxUint32, http2.ErrCodeNo, []byte{}); err != nil {
```

Comment:
It's possible we want to do some refactoring here.  I think I was expecting we'd pass an `error` to `http2Client.Close`, and that would dictate the code used here -- OR, pass the code in directly.  Anywhere we already send a GOAWAY, we could delete that and rely on this to do it instead.  Or we need a way to call `Close` to indicate a GOAWAY was already sent, and so `Close` should not send another one.